### ISSUE TITLE
[Bug]: PySide6 6.4.0 and FigureCanvasQTAgg Combability 

### ISSUE THREAD

**Issue by MosGeo** [matplotlib/matplotlib#24175]
### Bug summary

Using the latest PySide6 version (6.4.0) and Matplotlib version (3.6.1), it is not possible to import ```FigureCanvasQTAgg```. It seems that the whole Qt backend breaks.

### Code for reproduction

```python
from matplotlib.backends.backend_qtagg import FigureCanvasQTAgg
```

### Actual outcome

```
  File "C:\Users\Mustafa\Desktop\Projects\test_project\test.py", line 1, in <module>
    from matplotlib.backends.backend_qtagg import FigureCanvasQTAgg
  File "C:\Users\Mustafa\Desktop\Projects\test_project\.venv\lib\site-packages\matplotlib\backends\backend_qtagg.py", line 12, in <module>
    from .backend_qt import QtCore, QtGui, _BackendQT, FigureCanvasQT
  File "C:\Users\Mustafa\Desktop\Projects\test_project\.venv\lib\site-packages\matplotlib\backends\backend_qt.py", line 72, in <module>
    _MODIFIER_KEYS = [
  File "C:\Users\Mustafa\Desktop\Projects\test_project\.venv\lib\site-packages\matplotlib\backends\backend_qt.py", line 73, in <listcomp>
    (_to_int(getattr(_enum("QtCore.Qt.KeyboardModifier"), mod)),
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'KeyboardModifier'
```

### Expected outcome

Success import and usage

### Operating system

Windows

### Matplotlib Version

3.6.1

### Python version

3.10.8

### Installation

pip

**Comment by MosGeo** [comment]
Ok. Looks like it is fixed in the latest dev version.

**Comment by cyz1cyz** [comment]
Today(Oct 28) I also find this bug in pyside6 6.4.0.1

**Comment by zlmturnout** [comment]
> Today(Oct 28) I also find this bug in pyside6 6.4.0.1

same bug with python=3.11 matplotlib=3.6.1 and PySide6=6.4.0.1

**Comment by tacaswell** [comment]
The problem is an unexpected change in pyside6 in how they handled their enums.  This was fixed in #24158 and well be in mpl 3.6.2.
